roomserver/storage/shared/latest_events_updater.go

```diff
@@ -104,7 +112,9 @@ func (u *LatestEventsUpdater) HasEventBeenSent(eventNID types.EventNID) (bool, e
 
 // MarkEventAsSent implements types.RoomRecentEventsUpdater
 func (u *LatestEventsUpdater) MarkEventAsSent(eventNID types.EventNID) error {
-	return u.d.EventsTable.UpdateEventSentToOutput(u.ctx, u.txn, eventNID)
+	return u.d.Writer.Do(u.d.DB, nil, func(txn *sql.Tx) error {
+		return u.d.EventsTable.UpdateEventSentToOutput(u.ctx, u.txn, eventNID)
```

Comment:
Won't this make a txn in a txn?

Reply:
Good point, we should be giving `u.txn` to the `TransactionWriter` here.